tests/50federation/30room-join.pl

```diff
@@ -86,7 +80,18 @@ sub assert_is_valid_pdu {
             # that 404 response from SyTest and that fallback mechanism to test that the
```

Comment:
this comment seems to be somewhat incorrect.

Also: this change precludes testing against servers which don't support the v2 endpoint. Is it necessary? If so, Can you make the /v2/ request optional? (Just make a future, and cancel it once the v1 request arrives)